packages/fetch-engine/src/downloadZip.ts

```diff
@@ -17,28 +17,35 @@ const del = promisify(rimraf)
 
 export type DownloadResult = {
   lastModified: string
-  sha256: string
-  zippedSha256: string
+  sha256: string | null
+  zippedSha256: string | null
 }
 
-async function fetchSha256(url: string): Promise<{ sha256: string; zippedSha256: string }> {
-  // We get a string like this:
-  // "3c82ee6cd9fedaec18a5e7cd3fc41f8c6b3dd32575dc13443d96aab4bd018411  query-engine.gz\n"
-  // So we split it by whitespace and just get the hash, as that's what we're interested in
-  const [zippedSha256, sha256] = [
-    (
-      await fetch(`${url}.sha256`, {
-        agent: getProxyAgent(url) as any,
-      }).then((res) => res.text())
-    ).split(/\s+/)[0],
-    (
-      await fetch(`${url.slice(0, url.length - 3)}.sha256`, {
-        agent: getProxyAgent(url.slice(0, url.length - 3)) as any,
-      }).then((res) => res.text())
-    ).split(/\s+/)[0],
-  ]
-
-  return { sha256, zippedSha256 }
+async function fetchChecksum(url: string): Promise<string | null> {
+  try {
+    const checksumUrl = `${url}.sha256`
+    const response = await fetch(checksumUrl, {
+      agent: getProxyAgent(url) as any,
+    })
+    if (!response.ok) {
+      throw new Error(`Failed to fetch sha256 checksum at ${checksumUrl}. ${response.status} ${response.statusText}`)
+    }
+    const body = await response.text()
+    // We get a string like this:
+    // "3c82ee6cd9fedaec18a5e7cd3fc41f8c6b3dd32575dc13443d96aab4bd018411  query-engine.gz\n"
+    // So we split it by whitespace and just get the hash, as that's what we're interested in
+    const [checksum] = body.split(/\s+/)
+    if (!/^[a-f0-9]{64}$/gi.test(checksum)) {
+      throw new Error(`Unable to parse checksum from response body for ${checksumUrl}`)
```

Comment:
```suggestion
      throw new Error(`Unable to parse checksum from ${checksumUrl} - response body: ${body}`)
```
Would probably help debug if this happens